# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/chromium/hotfix-xkb-pkgconfig.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- chromium-151.0.7922.71/ui/events/keycodes/BUILD.gn.vanilla
+++ chromium-151.0.7922.71/ui/events/keycodes/BUILD.gn
@@ -4,17 +4,28 @@

 import("//build/config/ozone.gni")
 import("//build/config/ui.gni")
+import("//build/config/linux/pkg_config.gni")
 import("//ui/base/ui_features.gni")

+if (use_xkbcommon) {
+  pkg_config("xkbcommon") {
+    packages = [ "xkbcommon" ]
+  }
+}
+
 source_set("xkb") {
   sources = [
     "keyboard_code_conversion_xkb.cc",
     "keyboard_code_conversion_xkb.h",
     "scoped_xkb.h",
     "xkb_keysym.h",
   ]

+  if (use_xkbcommon) {
+    public_configs = [ ":xkbcommon" ]
+  }
+
   public_deps = [
     "//ui/base:buildflags",
     "//ui/gfx/x/keysyms",
   ]
